testing/test_doctest.py

```diff
@@ -475,6 +475,85 @@ def foo():
                                     "--junit-xml=junit.xml")
         reprec.assertoutcome(failed=1)
 
+    def _run_doctest_report(self, testdir, format):
+        testdir.makepyfile("""
+            def foo():
+                '''
+                >>> foo()
+                   a  b
+                0  1  4
+                1  2  4
+                2  3  6
+                '''
+                print('   a  b\\n'
+                      '0  1  4\\n'
+                      '1  2  5\\n'
+                      '2  3  6')
+            """)
+        return testdir.runpytest("--doctest-modules", "--doctest-report", format)
+
+    def test_doctest_report_udiff(self, testdir, format='udiff'):
+        result = self._run_doctest_report(testdir, format)
+        result.stdout.fnmatch_lines([
+            '     0  1  4',
+            '    -1  2  4',
+            '    +1  2  5',
+            '     2  3  6',
+        ])
+
+    def test_doctest_report_cdiff(self, testdir):
+        result = self._run_doctest_report(testdir, 'cdiff')
+        result.stdout.fnmatch_lines([
+            '         a  b',
+            '      0  1  4',
+            '    ! 1  2  4',
+            '      2  3  6',
+            '    --- 1,4 ----',
+            '         a  b',
+            '      0  1  4',
+            '    ! 1  2  5',
+            '      2  3  6',
+        ])
+
+    def test_doctest_report_ndiff(self, testdir):
+        result = self._run_doctest_report(testdir, 'ndiff')
+        result.stdout.fnmatch_lines([
+            '         a  b',
+            '      0  1  4',
+            '    - 1  2  4',
+            '    ?       ^',
+            '    + 1  2  5',
+            '    ?       ^',
+            '      2  3  6',
+        ])
+
+    def test_doctest_report_none_or_only_first_failure(self, testdir):
```

Comment:
e229a27 was about `test_doctest_report_udiff` correct? I mean here that you can avoid the inner loop and use `parametrize` in here as well:

```python
@pytest.mark.parametrize('format', ['none', 'only_first_failure'])
def test_doctest_report_none_or_only_first_failure(self, testdir):    
    result = self._run_doctest_report(testdir, format)
    result.stdout.fnmatch_lines([
        'Expected:',
        '       a  b',
        '    0  1  4',
        '    1  2  4',
        '    2  3  6',
        'Got:',
        '       a  b',
        '    0  1  4',
        '    1  2  5',
        '    2  3  6',
    ])
```